cmake/Modules/AddCFlagIfSupported.cmake

```diff
@@ -7,7 +7,7 @@ INCLUDE(CheckCCompilerFlag)
 
 MACRO(ADD_C_FLAG _FLAG)
 	STRING(TOUPPER ${_FLAG} UPCASE)
-	STRING(REGEX REPLACE "^-" "" UPCASE_PRETTY ${UPCASE})
+	STRING(REGEX REPLACE "-" "_" UPCASE_PRETTY ${UPCASE})
```

Comment:
I've given this a local spin, and I'd suggest to make it `REPLACE "[-=]"`.

The rationale is that, in some cases, the cached variable ends up as 
```
//Test IS__FVISIBILITY=HIDDEN_SUPPORTED
IS__FVISIBILITY:UNINITIALIZED=HIDDEN_SUPPORTED:INTERNAL=1

//Test IS__WIMPLICIT_FALLTHROUGH=1_SUPPORTED
IS__WIMPLICIT_FALLTHROUGH:UNINITIALIZED=1_SUPPORTED:INTERNAL=
```
which seems to completely confuse CMake on regeneration, so it always does the compile check. In the case of the Xcode generator, that's around 1-2s per test.

Reply:
Also, since AFAIU the point of that replace was to drop the initial `-` from the argument so it can be used as a variable, all `IS_.*_SUPPORTED` now get an extraneous `_` (as my exerpt ⬆️ shows). That's minor though, but you might want to drop the (now duplicated) `_` from the `IS_` strings.